app/services/workflow_service/property_assessment.rb

```diff
@@ -1,84 +1,21 @@
 module WorkflowService
   class PropertyAssessment < BaseWorkflowService
     def call
+      @remaining_mortgage_allowance ||= Threshold.value_for(:property_maximum_mortgage_allowance, at: @submission_date)
```

Comment:
Defining `@remaining_mortgage_allowance` and then using it `calculate_property` unnecessarily couples the two methods. It would be simplest to remove this line and then update `calculate_property to:
```ruby
    def calculate_property
      remaining_mortgage_allowance = Threshold.value_for(:property_maximum_mortgage_allowance, at: @submission_date)
      Property.transaction do
        capital_summary.additional_properties.each do |property|
          property.assess_equity!(remaining_mortgage_allowance)
          remaining_mortgage_allowance -= property.allowable_outstanding_mortgage
        end
        capital_summary.main_home&.assess_equity!(remaining_mortgage_allowance)
      end
    end
```
And if you modify `BaseWorkflowService` as I've suggested, `@submission_date` should be replaced with `submission_date`

Reply:
Done